PeerioTechnologies/peerio-desktop
# feat: "untrust device" checkbox in sign out dialog

Checkbox will only be shown with the new 2fa, and only if the user trusted the device during 2fa confirmation.

I also made the dialog appear always, not only when there's autologin enabled, since even without autologin, the user should be able to mark device as untrusted. I could make it appear only if the checkbox for it would appear, but I'm not sure such difference in behavior would be a good idea.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  

https://peerio.tpondemand.com/entity/7630

#### Linked PRs in other repos

Depends on: https://github.com/PeerioTechnologies/peerio-icebear/pull/71

#### Manual testing instructions  

Tests will only work on labs currently.

@ebilg do not test until I give it a go :)

A.

1. Click Logout on an account without 2fa enabled or with device not marked as trusted.
2. Observe that there's no checkbox on logout dialog.

B.

1. Click Logout on an account with 2fa enabled and with device marked as trusted.
2. Observe that there's an unchecked checkbox for untrusting the device on the logout dialog.
3. Do not check the checkbox, click Logout.
4. Log in again and verify that you're not asked for 2fa code.

C.

1-2. Same as B.
3. Check the "untrust device" checkbox, click Logout.
4. Log in again and verify that you *are* asked for 2fa code.

#### Additional notes
... 

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner
